debian/kafka-connect-base/include/etc/confluent/docker/healthcheck.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+if [[ -z $CONNECT_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM ]]; then
+  if [[ $(curl -s -o /dev/null -w %{http_code} http://$CONNECT_REST_ADVERTISED_HOST_NAME:$CONNECT_REST_PORT/) = 200 ]]; then
+    echo "Woohoo! Kafka Connect is up!"
+    exit 0
+  else 
+    echo -e $(date) "\tKafka Connect HTTP state: " $(curl -s -o /dev/null -w %{http_code} $CONNECT_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM://$CONNECT_REST_ADVERTISED_HOST_NAME:$CONNECT_REST_PORT/info) " (waiting for 200)"
+    exit 1
+  fi
+else
+  if [[ $(curl -k -s -o /dev/null -w %{http_code} $CONNECT_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM://$CONNECT_REST_ADVERTISED_HOST_NAME:$CONNECT_REST_PORT/) = 200 ]]; then
+    echo "Woohoo! Kafka Connect with SSL listener is up!"
+    exit 0
+  else 
+    echo -e $(date) "\tKafka Connect with SSL listener HTTP state: " $(curl -k -s -o /dev/null -w %{http_code} http://$CONNECT_REST_ADVERTISED_HOST_NAME:$CONNECT_REST_PORT/info) " (waiting for 200)"
+    exit 1
+  fi
+fi
```

Comment:
BTW, should we check if a connector was running or not? the REST endpoint will continue to work once the worker comes up. If any start up step fails (status topics not available, for example), I think the worker should die immediately.

Reply:
Not sure I follow - do you mean check for an actual connector? As a base image there'll not be a connector created a lot of the time.

Reply:
In that case, yes, we won't check the connectors, but this healthcheck is inherited by the connect and replicator images so they would benefit from it (once of course connectors are instantiated)?

Reply:
Yes they would benefit from that. Sounds like a v2 of the healthcheck :) For now this is just basic functionality to check the REST API.